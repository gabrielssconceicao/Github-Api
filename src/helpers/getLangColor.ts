export const languageColors: Record<string, string> = {
  JavaScript: '#f7df1e',
  Python: '#3572A5',
  Java: '#b07219',
  'C++': '#f34b7d',
  C: '#555555',
  'C#': '#178600',
  PHP: '#4F5D95',
  TypeScript: '#2b7489',
  Swift: '#ffac45',
  Ruby: '#CC342D',
  Go: '#00ADD8',
  R: '#198CE7',
  Kotlin: '#F18E33',
  Scala: '#c22d40',
  Rust: '#dea584',
  Dart: '#00B4AB',
  Perl: '#0298c3',
  Lua: '#000080',
  Matlab: '#e16737',
  Haskell: '#5e5086',
  SQL: '#e38c00',
  HTML: '#e34c26',
  CSS: '#563d7c',
  'Objective-C': '#438eff',
  Assembly: '#6E4C13',
  Shell: '#89e051',
  PowerShell: '#012456',
  Groovy: '#e69f56',
  Vue: '#41B883',
  Dockerfile: '#384d54',
  Clojure: '#db5855',
  Elixir: '#6e4a7e',
  D: '#ba595e',
  Julia: '#a270ba',
  Arduino: '#bd79d1',
  Verilog: '#b2b7f8',
  VHDL: '#adb2cb',
  Ada: '#02f88c',
  Scheme: '#1e4aec',
  'Objective-C++': '#6866fb',
  Crystal: '#000100',
};
export const getLangColor = (lang: string = ''): string => {
  let language = '#ffffff';
  if (lang in languageColors) {
    language = languageColors[lang];
  }
  return language;
};
