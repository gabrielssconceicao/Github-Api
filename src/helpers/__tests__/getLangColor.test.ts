import { describe, it, expect } from 'vitest';
import { getLangColor, languageColors } from '../getLangColor';

describe('getLangColor', () => {
  it('should return the correct color for the given language', () => {
    const color = getLangColor('JavaScript');
    expect(languageColors.JavaScript).toBe(color);

    const color2 = getLangColor('Python');
    expect(languageColors.Python).toBe(color2);

    const color3 = getLangColor('Java');
    expect(languageColors.Java).toBe(color3);

    expect(getLangColor('JavaScript')).toMatchSnapshot(
      'getLangColor("JavaScript")',
    );
    expect(getLangColor('Python')).toMatchSnapshot("getLangColor('Python')");
    expect(getLangColor('Java')).toMatchSnapshot("getLangColor('Java')");
  });

  it('should return the default color for an unknown language', () => {
    const color = getLangColor('Unknown');
    expect(color).toBe('#ffffff');
    expect(getLangColor('Unknown')).toMatchSnapshot("getLangColor('Unknown')");
  });
});
