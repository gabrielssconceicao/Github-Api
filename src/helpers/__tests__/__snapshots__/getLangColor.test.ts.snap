// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getLangColor > should return the correct color for the given language > getLangColor("JavaScript") 1`] = `"#f7df1e"`;

exports[`getLangColor > should return the correct color for the given language > getLangColor('Java') 1`] = `"#b07219"`;

exports[`getLangColor > should return the correct color for the given language > getLangColor('Python') 1`] = `"#3572A5"`;

exports[`getLangColor > should return the default color for an unknown language > getLangColor('Unknown') 1`] = `"#ffffff"`;
